js/views/conversation_view.js

```diff
@@ -2186,12 +2194,183 @@
       }
     },
 
+    handleDeleteOrBackspace(event, isDelete) {
+      const $input = this.$messageField[0];
+      const text = this.$messageField.val();
+
+      // Only handle the case when nothing is selected
+      if ($input.selectionDirection !== 'none') {
+        // Note: if this ends up deleting a handle, we should
+        // (ideally) check if we need to update the mapping in
+        // `this.memberView`, but that's not vital as we already
+        // reset it on every 'send'
+        return;
+      }
+
+      const mentions = this.memberView.pendingMentions();
+
+      const _ = window.Lodash; // no underscore.js please
+      const predicate = isDelete ? _.startsWith : _.endsWith;
+
+      const pos = $input.selectionStart;
+      const part = isDelete ? text.substr(pos) : text.substr(0, pos);
+
+      let curMention = null;
+      _.forEach(mentions, (val, key) => {
```

Comment:
This looks boilerplatey, is there any other way we could do the same?
Naively I would think along those lines?
```js
curMention = mentions.find(key => predicate(part,  key));
```

Reply:
This needed `_.keys()` as well as `find`. (I think I wanted early exit when I wrote this, but we shouldn't worry about performance here of course!) Updated accordingly.

Reply:
Ok I didn't realise `mentions` is an object :) changes look good!